AwardsServer/AwardsServer/ServerUI/UIForm.cs

```diff
@@ -65,11 +66,171 @@ public void UpdateWinners()
             }
         }
 
+        private struct OptionHold
+        {
+            public string VariableName;
+            public string AttributeValue;
+
+            public Control InputControl;
+            public Label NameControl;
+
+            public FieldInfo FieldInfo;
+
+            public object Value
+            {
+                get
+                {
+                    if(InputControl is TextBox)
+                    {
+                        TextBox tt = (TextBox)InputControl;
+                        return tt.Text;
+                    } else if(InputControl is NumericUpDown)
+                    {
+                        NumericUpDown tt = (NumericUpDown)InputControl;
+                        return (int)tt.Value;
+                    } else if(InputControl is CheckBox)
+                    {
+                        CheckBox tt = (CheckBox)InputControl;
+                        return tt.Checked;
+                    } else
+                    {
+                        return null;
+                    }
+                }
+            }
+
+            public Type InputType;
+            public void Clear()
+            {
+                InputControl.Dispose();
+                NameControl.Dispose();
+            }
+        }
+        List<OptionHold> options = new List<OptionHold>();
+
+        public void UpdateOptions()
```

Comment:
This function will procedural generate the label and whichever control will be used for the user to input

- Boolean -> Checkbox
- Integer -> Numberic up down thing
- String -> Text box

Reply:
gorgeous